cosmpy/aerial/client/__init__.py

```diff
@@ -716,3 +741,57 @@ def broadcast_tx(self, tx: Transaction) -> SubmittedTx:
         initial_tx_response.ensure_successful()
 
         return SubmittedTx(self, tx_digest)
+
+    def query_latest_block(self) -> Block:
+        """Query the latest block.
+
+        :return: latest block
+        """
+        req = GetLatestBlockRequest()
+        resp = self.tendermint.GetLatestBlock(req)
+        return self._parse_block(resp)
+
+    def query_block(self, height: int) -> Block:
+        """Query the block.
+
+        :param height: block height
+        :return: block
+        """
+        req = GetBlockByHeightRequest(height=height)
+        resp = self.tendermint.GetBlockByHeight(req)
+        return self._parse_block(resp)
+
+    def _parse_timestamp(self, timestamp: Timestamp) -> datetime:
+        """Parse the timestamp.
+
+        :param timestamp: timestamp
+        :return: parsed timestamp
+        """
+        return datetime.fromtimestamp(timestamp.seconds)
+
+    def _parse_block(self, block: GetBlockByHeightResponse) -> Block:
+        """Parse the block.
+
+        :param block: block as GetBlockByHeightResponse
+        :return: parsed block as Block
+        """
+        return Block(
+            height=int(block.block.header.height),
+            time=self._parse_timestamp(block.block.header.time),
+            tx_hashes=[sha256(tx).hex().upper() for tx in block.block.data.txs],
```

Comment:
We can get Tx here, not the tx response 

- Tx is in bytes format which can be deserialised into Tx protobuf object - but it has no hash value anyway.
The only solution for txhash I found is to just fo sha256 - You can query node using this hash to get TxResponse.

Reply:
The best solution would be to create dedicated method for Tx hash derivation in comspy, whic would be able to take as input parameter of various types (e.g. `bytes` (=Tx serialised in protobuf encoding) and  `Tx` & `TxRaw` python types generated from CosmosSDK protobuf schema (`github.com/cosmos/cosmos-sdk/types/tx`).

But for now, th in-place solution provided here is sufficient.